USE_CASES.md

```diff
@@ -1000,3 +1001,66 @@ namespace TwilioTest
     }
 }
 ```
+
+<a name="working-with-permissions" ></a>
+# Working with permissions
+
+The permissions builder is a convenient way to manipulate API key permissions when creating new API keys or managing existing API keys.
+You can use the types named according to the various [permissions](https://sendgrid.api-docs.io/v3.0/api-key-permissions) to add the scopes required for those permissions.
+For example, to create an API key for Alerts and Stats:
+
+```
+var apiKey = Environment.GetEnvironmentVariable("NAME_OF_THE_ENVIRONMENT_VARIABLE_FOR_YOUR_SENDGRID_KEY");
+var client = new SendGridClient(apiKey);
+var builder = new SendGridPermissionsBuilder();
+builder.AddPermissionsFor<Alerts>();
+builder.AddPermissionsFor<Stats>();
+
+var data = new
+{
+    name = "Alerts & Stats API Key",
+    scopes = builder.Build()
+};
+
+var json = JsonConvert.SerializeObject(data);
+await client.RequestAsync(SendGridClient.Method.POST, urlPath: "api_keys", requestBody: json);
+```
+
+There are also some extension methods to easily create API keys for various common use cases. 
+For example, to create a read-only mail send API key that can be used only to send emails:
+
+```
+var apiKey = Environment.GetEnvironmentVariable("NAME_OF_THE_ENVIRONMENT_VARIABLE_FOR_YOUR_SENDGRID_KEY");
+var client = new SendGridClient(apiKey);
+var builder = new SendGridPermissionsBuilder();
+builder.CreateReadOnlyMailSend();
```

Comment:
It might be nice if we had:
```csharp
builder.CreateReadOnlyMailSend();
client.Request(builder);
```
but I think that's out of scope for this PR. This is a note to self to file a follow up issue.